pkg/controller/provider/container/vsphere/watch.go

```diff
@@ -181,25 +181,33 @@ func (r *VMEventHandler) search() {
 		Log.Trace(err)
 		return
 	}
-	list := []model.VM{}
-	err = r.DB.List(
-		&list,
-		libmodel.ListOptions{
-			Predicate: libmodel.Or(
-				libmodel.Neq("Revision", libmodel.Field{Name: "RevisionValidated"}),
-				libmodel.Neq("PolicyVersion", version)),
-		})
-	if err != nil {
-		Log.Trace(liberr.Wrap(err))
-		return
-	}
-	for _, vm := range list {
-		if revision, found := r.reported[vm.ID]; found {
-			if vm.Revision == revision {
-				continue
+	for {
+		list := []model.VM{}
+		err = r.DB.List(
+			&list,
+			libmodel.ListOptions{
+				Predicate: libmodel.Or(
+					libmodel.Neq("Revision", libmodel.Field{Name: "RevisionValidated"}),
+					libmodel.Neq("PolicyVersion", version)),
+				Page: &libmodel.Page{
+					Limit: 250,
```

Comment:
I think you might be missing an offset here.

Reply:
The offset (default to 0) is intentional.  Each pass will validate VM which will be excluded from the next query because validated VMs will not match.

Reply:
Ah, understood. Thanks for explaining!